File: Java-Projects/JavaFeatures/src/main/java/my/learning/jf/task/SetIntersection.java
package my.learning.jf.task;

import java.util.HashSet;

public class SetIntersection {
    public static void main(String[] args) {
        /*
            j = qwerd s = qwerd - 5
            j = 123qwerd456 s = qwerd - 5
            j = 123 s = 456 - 0
            j = 1q2w3e s = qwe - 3
            j = qwe s = rqwet - 3
            one of them or both is empty - 0
            one of them or both is null - 0
         */
        System.out.println("j = qwerd s = qwerd - 5, res is " + count("qwerd", "qwerd"));
        System.out.println("j = 123qwerd456 s = qwerd - 5, res is " + count("123qwerd456", "qwerd"));
        System.out.println("j = 123 s = 456 - 0, res is " + count("123", "456"));
        System.out.println("j = 1q2w3e s = qwe - 3, res is " + count("1q2w3e", "qwe"));
        System.out.println("j = qwe s = rqwet - 3, res is " + count("qwe", "rqwet"));
        System.out.println("one of them or both is empty - 0, res is " + count("213", ""));
        System.out.println("one of them or both is empty - 0, res is " + count("", "213"));
        System.out.println("one of them or both is empty - 0, res is " + count("", ""));
        System.out.println("one of them or both is null - 0, res is " + count(null, "123"));
        System.out.println("one of them or both is null - 0, res is " + count("123", null));
        System.out.println("one of them or both is null - 0, res is " + count(null, null));
    }

    private static int count(String source, String potentialInclusions) {
        if ((source == null || source.isEmpty()) || (potentialInclusions == null || potentialInclusions.isEmpty())) {
            return 0;
        }

        var sourceSet = new HashSet<>(source.length());
        for (char c : source.toCharArray()) {
            sourceSet.add(c);
        }

        int count = 0;

        for (char c : potentialInclusions.toCharArray()) {
            if (sourceSet.contains(c)) count++;
        }

        return count;
    }

}

File: Python-Projects/Python/src/yndx/set_intersection.py
# def count(jewelery: str, stones: str):
#     if not jewelery or not stones:
#         return str(0)
#     return str(len(set(jewelery) & set(stones)))
#
#
# print("j = qwerd s = qwerd - 5, res is " + count("qwerd", "qwerd"))
# print("j = 123qwerd456 s = qwerd - 5, res is " + count("123qwerd456", "qwerd"))
# print("j = 123 s = 456 - 0, res is " + count("123", "456"))
# print("j = 1q2w3e s = qwe - 3, res is " + count("1q2w3e", "qwe"))
# print("j = qwe s = rqwet - 3, res is " + count("qwe", "rqwet"))
# print("one of them or both is empty - 0, res is " + count("213", ""))
# print("one of them or both is empty - 0, res is " + count("", "213"))
# print("one of them or both is empty - 0, res is " + count("", ""))
# print("one of them or both is null - 0, res is " + count(None, "123"))
# print("one of them or both is null - 0, res is " + count("123", None))
# print("one of them or both is null - 0, res is " + count(None, None))


import sys

jewelries = input().strip()
stones = input().strip()

print('Work')

if not jewelries or not stones:
    print(str(0))
else:
    print(str(len(set(jewelries) & set(stones))))

# exit(0)
#